cdhweb/events/models.py

```diff
@@ -413,4 +415,40 @@ class EventsLandingPage(StandardHeroMixinNoImage, Page):
     settings_panels = Page.settings_panels
 
     # allow content pages to be added under events for special event series
-    subpage_types = [Event, LinkPage, ContentPage]
+    subpage_types = [Event]
+
+    def get_upcoming_events_for_semester(self, semester, year):
+        # Adjust the semester and year to datetime ranges
+        if semester == "spring":
+            start_date = datetime.date(year, 1, 1)
+            end_date = datetime.date(year, 5, 31)
+        elif semester == "summer":
+            start_date = datetime.date(year, 6, 1)
+            end_date = datetime.date(year, 8, 31)
+        elif semester == "fall":
+            start_date = datetime.date(year, 9, 1)
+            end_date = datetime.date(year, 12, 31)
+        else:
+            raise ValueError("Invalid semester")
+
+        # Filter events based on start_time within the semester range
+        return (
+            self.get_children()
+            .live()
+            .filter(event__start_time__gte=start_date, event__start_time__lte=end_date)
+            .order_by("event__start_time")
+        )
+
+    def get_upcoming_events(self):
+        current_datetime = timezone.now()
+
+        child_pages = self.get_children().live()
+
+        # Fetch upcoming events among the child pages
+        return child_pages.filter(event__start_time__gte=current_datetime).order_by(
+            "event__start_time"
+        )
+    
+    def save(self, *args, **kwargs):
+        self.slug = 'events'
```

Comment:
This will _probably_ work, but get a bit dicey with revisions and things. I'd say leave it for now, but if it gives you trouble, you can follow [this pattern of overriding the page form](https://github.com/springload/brn-branz-website/blob/e3f965df9095a3c4b0596d126ef355f3520f4077/certificates/models.py#L23-L59)